VideoLocker/src/main/java/org/edx/mobile/authentication/LoginAPI.java

```diff
@@ -104,13 +102,20 @@ public AuthResponse logInUsingGoogle(String accessToken) throws Exception {
     @NonNull
     private AuthResponse finishSocialLogIn(@NonNull String accessToken, @NonNull LoginPrefs.AuthBackend authBackend) throws Exception {
         final String backend = ApiConstants.getOAuthGroupIdForAuthBackend(authBackend);
-        final AuthResponse response = loginService.exchangeAccessToken(accessToken, config.getOAuthClientId(), backend);
-        if (response.error != null && response.error.equals("401")) {
+        final Response<AuthResponse> response = loginService.exchangeAccessToken(accessToken, config.getOAuthClientId(), backend).execute();
+        if (response.code() == HttpURLConnection.HTTP_UNAUTHORIZED) {
             // TODO: Introduce a more explicit error code to indicate that an account is not linked.
             throw new AccountNotLinkedException();
         }
-        finishLogIn(response, authBackend, "");
-        return response;
+        if (!response.isSuccessful()) {
+            throw new HttpResponseStatusException(response.code());
+        }
+        final AuthResponse data = response.body();
+        if (data.error != null && data.error.equals(Integer.toString(HttpURLConnection.HTTP_UNAUTHORIZED))) {
```

Comment:
Yeah, since I had to throw the `HttpResponseStatusException` manually here (as Retrofit 2 doesn't have the automatic error handler for error codes), I felt it proper to throw the appropriate exception (instead of the generic one) in the case of receiving the 'Unauthorized' error as an HTTP error code instead of in the response, though it seems this wasn't done previously.